proparse/src/main/java/org/prorefactor/treeparser01/TP01Support.java

```diff
@@ -999,9 +1029,8 @@ public void recordNameNode(AST anode, ContextQualifier contextQualifier) throws
   }
 
   @Override
-  public void routineReturnDatatype(AST datatypeAST) {
-    JPNode datatypeNode = (JPNode) datatypeAST;
-    if (datatypeNode.getType() == NodeTypes.CLASS)
+  public void routineReturnDatatype(JPNode datatypeNode) {
+    if (datatypeNode.getType() == ProParserTokenTypes.CLASS)
       datatypeNode = datatypeNode.nextNode();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Introduce a new variable instead of reusing the parameter "datatypeNode". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1226)